dbt_subprojects/dex/macros/models/_project/uniswap_downstream_trades.sql

```diff
@@ -0,0 +1,371 @@
+{% macro uniswap_downstream_trades(
+    blockchain = null
+    , has_univ4 = null 
+    , has_bunni = null 
+    )
+%}
+
+with 
+
+v4_trades as (
+    {% if has_univ4|lower == 'yes' %}
```

Comment:
it may be better to have if statement prior to the CTE start, so that you can ignore the source read altogether if false and not force a row of nulls?
otherwise, you will have empty rows? or i guess the inner joins downstream filter out these nulls rows? either way, i think code wise it may be cleaner to have outside.

Reply:
hmm, yeah the inner join filters out the null rows! I'd need to add a lot `{ if has_univ4 }` where im picking the fees from because of the coalesce im using hence why i opted to have an empty row of nulls! they get ignored eventually. I'll update now so it's outside the CTE if that's cleaner!